store/api/views.py

```diff
@@ -0,0 +1,165 @@
+from django_filters.rest_framework import DjangoFilterBackend
+from drf_yasg import openapi
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import generics, viewsets, status
+from rest_framework.generics import RetrieveUpdateAPIView
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from store.api.filters import ProductFilter
+from store.api.permissions import IsAdmin
+from store.models import Product, Category
+from store.api.serializers import (
+    CategorySerializer,
+    ProductStaffSerializer,
+    ProductSerializer,
+)
+
+
+class ProductSearchViewSet(viewsets.ReadOnlyModelViewSet):
+    queryset = Product.objects.all()
+    serializer_class = ProductSerializer
+    filter_backends = (DjangoFilterBackend,)
+    filterset_class = ProductFilter
+
+    CATEGORY = openapi.Parameter(
+        name="category",
+        in_=openapi.IN_QUERY,
+        description="Filter products by category. Use commas to specify multiple categories.",
+        type=openapi.TYPE_STRING,
+    )
+    MIN_PRICE = openapi.Parameter(
+        name="min_price",
+        in_=openapi.IN_QUERY,
+        description="Filter products by minimum price.",
+        type=openapi.TYPE_NUMBER,
+    )
+    MAX_PRICE = openapi.Parameter(
+        name="max_price",
+        in_=openapi.IN_QUERY,
+        description="Filter products by maximum price.",
+        type=openapi.TYPE_NUMBER,
+    )
+    NAME = openapi.Parameter(
+        name="name",
+        in_=openapi.IN_QUERY,
+        description="Filter products by name. Search is case-insensitive.",
+        type=openapi.TYPE_STRING,
+    )
+
+    @swagger_auto_schema(manual_parameters=[CATEGORY, MIN_PRICE, MAX_PRICE, NAME])
+    def list(self, request, *args, **kwargs):
+        return super().list(request, *args, **kwargs)
+
+
+class ProductStuffViewSet(viewsets.ModelViewSet):
+    serializer_class = ProductStaffSerializer
+    queryset = Product.objects.all()
+
+
+class ProductCreateAPIView(APIView):
+    permission_classes = (IsAdmin, IsAuthenticated)
+    """
+    API endpoint for creating a new product.
+    """
+
+    @swagger_auto_schema(
+        operation_description="API endpoint for creating a new product.",
+        request_body=ProductStaffSerializer,
+        responses={201: "Product successfully created"},
```

Comment:
I think you should use smth like:
```suggestion
        responses={201: openapi.Response('Product', ProductSerializer)},
```